sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1356,4 +1360,36 @@ export class ApplicationService extends RecordDataModelService<Application> {
       }
     }
   }
+
+  /**
+   * Archives one or more applications when 43 days
+   * have passed the end of the study period.
+   * @param auditUserId user making changes to table.
+   */
+  async archiveApplications(auditUserId: number): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+
+    // Build sql statement to get all application ids to archive
+    const applicationsToArchive = this.repo
+      .createQueryBuilder("application")
+      .select("application.id")
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .where("application.applicationStatus = :completed")
+      .andWhere(
+        "(CURRENT_DATE - offering.studyEndDate) > :applicationArchiveDays",
+      )
+      .andWhere("application.isArchived <> :isApplicationArchived")
+      .getSql();
+
+    this.repo
+      .createQueryBuilder()
+      .update(Application)
+      .set({ isArchived: true, modifier: auditUser })
+      .where(`applications.id IN (${applicationsToArchive})`)
```

Comment:
Please test this scenario with an empty array. I believe it won't fail. I saw one of these old issues mentioning it. 
https://github.com/typeorm/typeorm/issues/3630

Reply:
Thanks for this, I will test.

Reply:
No need to test it. It will work in the subquery, like below:
```
select * from sims.applications a where a.id in (select id from sims.applications a2 where 1 = 2)
```